tool/internal/ast/shared.go

```diff
@@ -44,6 +44,48 @@ func FindFuncDeclWithoutRecv(root *dst.File, funcName string) *dst.FuncDecl {
 	return decls[0]
 }
 
+// extractBaseReceiverType extracts the base type name from a receiver expression,
+// handling both generic and non-generic types.
+// For example:
+// - *MyStruct -> *MyStruct
+// - MyStruct -> MyStruct
+// - *GenStruct[T] -> *GenStruct
+// - GenStruct[T] -> GenStruct
+func extractBaseReceiverType(recvTypeExpr dst.Expr) string {
```

Comment:
How about renaming it to "stripGenericTypes"? It might be more general. But it's up to you; the current name is fine as well.